src/chttpd/src/chttpd_db.erl

```diff
@@ -481,18 +482,62 @@ db_req(#httpd{method='POST', path_parts=[_, <<"_bulk_get">>]}=Req, Db) ->
             } = bulk_get_parse_doc_query(Req),
             Options = [{user_ctx, Req#httpd.user_ctx} | Options0],
 
-            {ok, Resp} = start_json_response(Req, 200),
-            send_chunk(Resp, <<"{\"results\": [">>),
-
-            lists:foldl(fun(Doc, Sep) ->
-                {DocId, Results, Options1} = bulk_get_open_doc_revs(Db, Doc,
-                                                                    Options),
-                bulk_get_send_docs_json(Resp, DocId, Results, Options1, Sep),
-                <<",">>
-            end, <<"">>, Docs),
-
-            send_chunk(Resp, <<"]}">>),
-            end_json_response(Resp)
+            % Decide whether to use multipart response or application/json
+            AcceptMixedMp = MochiReq:accepts_content_type("multipart/mixed"),
+            AcceptRelatedMp = MochiReq:accepts_content_type("multipart/related"),
+            AcceptMp = AcceptMixedMp orelse AcceptRelatedMp,
```

Comment:
Sorry one more thing I realized. I think this logic would make multipart response become the default if there is no accept header or it accepts '*/*'. 

In general the semantics would be correct, but given that this is a new behavior, it would be better if in such cases we returned json, and only if json is not accepted but multipart is then we send multipart.

So I think we'd want something like:

```
AcceptJson =  MochiReq:accepts_content_type("application/json"),
...
SendMp = not AcceptJson andalso (AcceptMixedMp orelse AcceptRelatedMp)

case SendMp of ...
```

So we are saying we only send mp if json wasn't requested (either because of a */* accept header or explicitly) and one of the mp flavors was. Otherwise we send json as before.

Reply:
@nickva Valid point even though I consider this kind of scenario a rare edge case. I'll make appropriate changes.

Reply:
@nickva Actually this change broke my tests with `got invalid Content-Type 'application/json'`
Have to investigate why.

Reply:
For some reason `MochiWeb`'s `accepts_content_type("application/json")` returns `true` even though the client sent these headers:
```
Accept = "multipart/related";
"Content-Type" = "application/json";
```
With previous logic it works perfectly because we have an explicit check for multipart `Accept`.

Reply:
@nickva Also I see the same logic in the normal db/doc request [here](https://github.com/apache/couchdb/blob/master/src/chttpd/src/chttpd_db.erl#L756) where we first just check for `multipart` `Accept` header. When i wrote this implementation for `_bulk_get` I mostly copied `db_doc_req` logic.

Reply:
I think the difference is that with existing open_revs logic it's ok if it is inconsistent because it is already release, but I am worried about changing the default behavior of what _bulk_gets if there is no `Accept` header or `Accept` header contains `*/*` (that is accept anything).

In those cases I think `_bulk_get` will start returning mp content instead of json as before. Try for example sending a request before the PR to _bulk_get without an Accept header and after the PR. For me, it looked like the default changed and that might be unexpected (even if technically correct from the HTTP request / response point).

Reply:
@nickva Yeah you are right here. Sorry for my unjustified stubbornness. I pushed a fix and squashed all the commits. Also the branch is rebased again.